torch/csrc/jit/script/python_sugared_value.cpp

```diff
@@ -525,6 +525,15 @@ std::shared_ptr<SugaredValue> toSugaredValue(
     auto& pyCu = CompilationUnit::_get_python_cu();
     if (auto classType = pyCu.get_class(c10::QualifiedName(qualifiedName))) {
       return std::make_shared<ClassValue>(classType);
+    } else if (getRecursiveScriptMode()) {
+      // Register class
+      auto rcb = py::module::import("torch._jit_internal").attr("createResolutionCallback")(2);
```

Comment:
This is not the right scope. This is the scope of the function being called. It is also really really fragile because it is so deep in the compiler and hardcoding the magic 2. This needs to use scope in which the class is defined. Since we are enforcing that class must have a qualifiedName, then we know the module in which it was defined. That module's global scope is probably the correct scope.